src/main/java/seedu/algobase/logic/commands/EditCommand.java

```diff
@@ -93,7 +93,12 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         model.setProblem(problemToEdit, editedProblem);
-        model.addTags(editProblemDescriptor.tags);
+        // model.addTags(editProblemDescriptor.tags);
+        if (editProblemDescriptor.tags != null) {
+            for (Tag tag : editProblemDescriptor.tags) {
+                model.addTag(tag);
```

Comment:
Consider changing the logic to: first check `model.hasTag()` then `addTag` if the tag doesn't exist.